extensions/vertx-http/runtime/src/main/java/io/quarkus/devui/runtime/LocalHostOnlyFilter.java

```diff
@@ -0,0 +1,50 @@
+package io.quarkus.devui.runtime;
+
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+
+import org.jboss.logging.Logger;
+
+import io.vertx.core.Handler;
+import io.vertx.core.http.HttpServerResponse;
+import io.vertx.ext.web.RoutingContext;
+
+public class LocalHostOnlyFilter implements Handler<RoutingContext> {
+    private static final Logger LOG = Logger.getLogger(LocalHostOnlyFilter.class);
+
+    private static final String LOCAL_HOST = "localhost";
+    private static final String LOCAL_HOST_IP = "127.0.0.1";
```

Comment:
Need to support IPv6 (`::1`)

Reply:
I can add that

Reply:
I can't help feel like we are somehow circumventing or not utilizing the right cors setup if we have to do these tricks? how come this is not caught by normal cors?

Reply:
This is not really CORS. It's allowed hosts. In the case if fails the origin header is null, so CORS can not work